pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -186,6 +409,185 @@ func (s *DiscoveryServer) updateCluster(push *model.PushContext, clusterName str
 	return nil
 }
 
+// SvcUpdate is a callback from service discovery to update the port mapping.
+func (s *DiscoveryServer) SvcUpdate(cluster, hostname string, ports map[string]uint32, rports map[uint32]string) {
+	pc := s.globalPushContext()
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	if cluster == "" {
+		pc.ServiceName2Ports[hostname] = ports
+		pc.ServicePort2Name[hostname] = rports
+	}
+	// TODO: for updates from other clusters, warn if they don't match primary.
+}
+
+// Update clusters for an incremental EDS push, and initiate the push.
+// Only clusters that changed are updated/pushed.
+func (s *DiscoveryServer) edsIncremental(version string, push *model.PushContext, edsUpdates map[string]*model.ServiceShards) {
+	adsLog.Infof("XDS:EDSInc Pushing %s Services: %v, "+
+		"VirtualServices: %d, ConnectedEndpoints: %d", version, edsUpdates,
+		len(push.VirtualServiceConfigs), adsClientCount())
```

Comment:
can you also print the destinationrules?

Reply:
In this PR ? I haven't touched destinationrules. Probably we should add it.